File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        sep=max=(len(list(set(s))))
        ans=0
        found=False
        if max<=1 or max==len(s):
            return max
        while (found!=True):
            for i in range(0,len(s)-sep+1):
                test=s[i:i+sep]
                if len(set(test))==max:
                    ans =max
                    found=True
                    break

            sep-=1
            max-=1

        return  (ans)
        

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max=0;
        
        Map<Character,Integer> map=new HashMap<>();
        int start=0;

        for(int i=0;i<s.length();i++){
            char curr=s.charAt(i);

            if(map.containsKey(curr) && map.get(curr)>=start){
                start=map.get(curr)+1;

            }
            map.put(curr,i);
            System.out.println(i-start+1);
            max=Math.max(max,i-start+1);
        }
        


        return max;
    }
}